File: Hot 100/0049_GroupAnagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();
        for (String str : strs) {
            char[] chars = str.toCharArray();
            Arrays.sort(chars);
            String newStr = new String(chars);
            map.computeIfAbsent(newStr, k -> new ArrayList<>()).add(str);
        }
        return new ArrayList<>(map.values());
    }
}

File: Hot 100/Hashing/0049_group_anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        buckets: defaultdict[str, List[str]] = defaultdict(list);
        for s in strs:
            key = ''.join(sorted(s));
            buckets[key].append(s);
        return list(buckets.values());

